src/anniversary/sf33rd/Source/Game/efff6.c

```diff
@@ -0,0 +1,520 @@
+#include "sf33rd/Source/Game/efff6.h"
+#include "common.h"
+#include "sf33rd/Source/Game/CHARSET.h"
+#include "sf33rd/Source/Game/EFFECT.h"
+#include "sf33rd/Source/Game/OPENING.h"
+#include "sf33rd/Source/Game/color3rd.h"
+#include "sf33rd/Source/Game/debug/Debug.h"
+#include "sf33rd/Source/Game/ta_sub.h"
+#include "sf33rd/Source/Game/texcash.h"
+
+extern u32 _op_char_table[];
+
+// forward decls
+void efff6_move_common(WORK_Other *ewk);
+void efff6_move01(WORK_Other *ewk);
+
+void effect_F6_move(WORK_Other *ewk) {
+    switch (ewk->wu.routine_no[0]) {
+    case 0:
+        if (ewk->wu.old_rno[1] <= op_w.index) {
+            ewk->wu.routine_no[0] += 1;
+        } else if (ewk->wu.old_rno[2] <= op_w.index) {
+            if (ewk->wu.type == 0x1B) {
+                efff6_move01(ewk);
+            } else {
+                efff6_move(ewk);
+            }
+        }
+        break;
+    default:
+        push_effect_work(&ewk->wu);
+    }
+}
+
+void efff6_move(WORK_Other *ewk) {
+    switch (ewk->wu.routine_no[2]) { /* irregular */
+    case 0:
+        ewk->wu.routine_no[2] += 1;
+        ewk->wu.disp_flag = 1;
+        set_char_move_init2(&ewk->wu, 0, ewk->wu.old_rno[0], ewk->wu.char_index, 0);
+        if (ewk->wu.my_mr.size.x == 0x3F) {
+            ewk->wu.my_mr_flag = 0;
+        } else {
+            ewk->wu.my_mr_flag = 1;
+        }
+        if (ewk->wu.type == 0x36) {
+            push_color_trans_req((ewk->wu.my_col_code & 0x1FF) + 0x3C, 8);
+            push_color_trans_req((ewk->wu.my_col_code & 0x1FF) + 0x3D, 9);
+            push_color_trans_req((ewk->wu.my_col_code & 0x1FF) + 0x3E, 10);
+            ewk->wu.my_col_code = 8;
+        } else {
+            push_color_trans_req(ewk->wu.my_col_code & 0x1FF, 0);
+            ewk->wu.my_col_code = 0;
+        }
+        /* fallthrough */
+    case 1:
+        efff6_move_common(ewk);
+        disp_pos_trans_entry5(ewk);
+    }
+}
+
+void efff6_move_common(WORK_Other *ewk) {
+    s16 work;
+
+    switch (ewk->wu.routine_no[3]) {
+    case 0:
+        switch (ewk->wu.direction) { /* switch 1; irregular */
+        case 0x20:                   /* switch 1 */
+            add_x_sub(ewk);
+            if (ewk->wu.xyz[0].disp.pos <= ewk->wu.old_rno[3]) {
+                ewk->wu.routine_no[3] += 1;
+                ewk->wu.xyz[0].disp.pos = ewk->wu.old_rno[3];
+            }
+            /* fallthrough*/
+        default:
+            break;
+        case 0x10: /* switch 1 */
```

Comment:
These should be removed 
```suggestion
        case 0x10:
```